application/models/services/QuestionAttributeHelper.php

```diff
@@ -209,4 +215,50 @@ public function fillMissingCategory($aAttributes, $sCategoryName)
         }
         return $aAttributes;
     }
+
+    /**
+     * This function returns an array of the attributes for the particular question
+     * including their values set in the database
+     *
+     * @param \Question $oQuestion  The question object
+     * @param string|null $sLanguage If you give a language then only the attributes for that language are returned
+     * @param string|null $sQuestionThemeOverride   Name of the question theme to use instead of the question's current theme
+     * @param boolean $advancedOnly If set to true, only the advanced attributes will be returned
+     * @return array
+     */
+    public function getQuestionAttributesWithValues($oQuestion, $sLanguage = null, $sQuestionThemeOverride = null, $advancedOnly = false)
+    {
+        $oSurvey = $oQuestion->survey;
+        if (empty($oSurvey)) {
+            throw new \Exception('This question has no survey - qid = ' . json_encode($oQuestion->qid));
```

Comment:
json_encode ?

Reply:
json_encode will make null = "null" instead of empty string.

Reply:
But : $oQuestion have always a qid (it's the pk). Maybe null value for sid ?

Maybe 

\Exception('Question " . $oQuestion->qid . ' survey  ' . json_encode($oQuestion->sid) . 'not found');

Reply:
+ sprintf ^^